apps/imaginary.sh

```diff
@@ -0,0 +1,153 @@
+    for app in "${APPS[@]}"
+    do
+        if is_this_installed "$app"
+        then
+            apt-get purge "$app" -y
+        fi
+    done
+    if is_this_installed ffmpeg && ! is_app_installed integration_whiteboard
+    then
+        apt-get purge ffmpeg -y
+    fi
+    apt-get autoremove -y
+    if yesno_box_yes "Do you want to remove all previews that were generated until now?
+This will most likely clear a lot of space! Also, pre-generated previews are not needed anymore once Imaginary are installed."
+    then
+        countdown "Removing the preview folder. This can take a while..." "5"
+        rm -rfv "$NCDATA"/appdata_*/preview
+        print_text_in_color "$ICyan" "Scanning Nextclouds appdata directory after removing all previews. \
+This can take a while..."
+        nextcloud_occ files:scan-app-data -vvv
+        msg_box "All previews were successfully removed."
+    fi
+    # Remove log
+    rm -f "$VMLOGS"/previewgenerator.log
+    # Show successful uninstall if applicable
+    removal_popup "$SCRIPT_NAME"
+fi
+
+# Generate docker-compose.yml
+mkdir -p "$SCRIPTS"/imaginary-docker
+if [ ! -f "$SCRIPTS/imaginary-docker/docker-compose.yml" ]
+then
+    touch "$SCRIPTS/imaginary-docker/docker-compose.yml"
+    cat << IMAGINARY_DOCKER_CREATE > "$SCRIPTS"/imaginary-docker/docker-compose.yml
+version: '3.1'
+services:
+  imaginary:
+    image: nextcloud/aio-imaginary:latest
+    container_name: imaginary
+    restart: always
+    environment:
+       PORT: 9000
+    command: -concurrency 50 -enable-url-source -log-level debug
+    ports:
+      -  127.0.0.1:9000:9000
+IMAGINARY_DOCKER_CREATE
+    print_text_in_color "$IGreen" "SCRIPTS/imaginary-docker/docker-compose.yml was successfully created."
+fi
+
+# Start the container
+docker compose -p imaginary -f "$SCRIPTS"/imaginary-docker/docker-compose.yml up -d
+
+# Test if imaginary is working
+countdown "Testing if it works in 3 sedonds" "3"
+if curl -O "http://127.0.0.1:9000/crop?width=500&height=400&url=https://raw.githubusercontent.com/h2non/imaginary/master/testdata/large.jpg"
+then
+    print_text_in_color "$IGreen" "imaginary seems to be working OK!"
+    rm -f large.jpg
+else
+    msg_box "Test failed, please report this to: $ISSUES"
+    exit
+fi
+
+# Install dependencies for Imaginary
+check_php
+install_if_not php"$PHPVER"-sysvsem
+
+# Set default limits
+# https://github.com/nextcloud/server/pull/18210/files#diff-3bbe91e1f85eec5dbd0031642dfb0ad6749b550fc3b94af7aa68a98210b78738R1121
+nextcloud_occ config:system:set preview_concurrency_all --value="8"
+nextcloud_occ config:system:set preview_concurrency_new --value="4"
+
+# Set providers (https://github.com/nextcloud/server/blob/master/lib/private/Preview/Imaginary.php#L60)
+# https://github.com/nextcloud/vm/pull/2464#discussion_r1155074227
+# This is handled by Imagniary itself
+nextcloud_occ config:system:set enabledPreviewProviders --value="OC\\Preview\\Imaginary"
```

Comment:
```suggestion
nextcloud_occ config:system:set enabledPreviewProviders 0 --value="OC\\Preview\\Imaginary"
```

Reply:
also you will need to add additionaly like the Movie Preview Provider and others that are not included in imaginary IIRC